dashboard/app/helpers/proxy_helper.rb

```diff
@@ -134,9 +134,10 @@ def resolve_redirect_url(location, allowed_hostname_suffixes:, redirect_limit: 5
   def allowed_hostname?(url, allowed_hostname_suffixes)
     return true unless allowed_hostname_suffixes
     return false unless url.hostname
-
+    hostname = url.hostname.downcase
     allowed_hostname_suffixes.find do |suffix|
-      url.hostname.downcase.match(Regexp.new("#{Regexp.escape(suffix)}$"))
+      hostname.match(Regexp.new("^#{Regexp.escape(suffix)}$")) ||
+        hostname.match(Regexp.new("\\.#{Regexp.escape(suffix)}$"))
```

Comment:
When you said "word boundary" I was worried you meant `/\b#{suffix}$/` which would match `evil-bar.org`, but you actually meant `/^#{suffix}$/` or `/\.#{suffix}$/` which I think is the right approach.  Actually, may I suggest `http://evil-bar.org` as an additional test case to refute?

It's not too bad to express this with just one regex; you can express "start of line or dot" with `(^|\.)`.
```ruby
hostname.match(Regexp.new("(^|\\.)#{Regexp.escape(suffix)}$"))
```